File: archives/C#/0121. Best Time to Buy and Sell Stock.cs
public class Solution {
    public int MaxProfit(int[] prices) {
        int rep = 0;
        if (prices.Length == 0)
        {
            return 0;
        }
        else
        {
            int minprice = prices[0];
            for (int i = 1; i < prices.Length; i++)
            {
                if (prices[i] < minprice)
                   { 
                    minprice = prices[i];
                }
                else
                {
                    rep = Math.Max(rep, prices[i] - minprice);
                }                                  
            }
                return rep;
            }
        }    
    }

File: archives/20190519python/0121. Best Time to Buy and Sell Stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        if not prices:
            return 0
        minPrice=prices[0]
        rep=0
        for price in prices:
            minPrice=min(minPrice,price)
            rep=max(rep,price-minPrice)
        return rep
